block/src/qcow/mod.rs

```diff
@@ -31,13 +31,16 @@ use vmm_sys_util::{
 
 pub use crate::qcow::raw_file::RawFile;
 
+/// Nesting depth limit for disk formats that can open other disk files.
+const MAX_NESTING_DEPTH: u32 = 10;
+
 #[sorted]
 #[derive(Debug, Error)]
 pub enum Error {
     #[error("Backing file io error: {0}")]
     BackingFileIo(io::Error),
     #[error("Backing file open error: {0}")]
-    BackingFileOpen(Box<crate::Error>),
+    BackingFileOpen(Box<Error>),
```

Comment:
Why does this error need to be `Box`ed? Because it embeds itself?

Reply:
Yes, it can't be nested without a `Box` or anything else that allocates it on the heap.